addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/FormHandlerParamField.java

```diff
@@ -89,18 +109,34 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * Checks whether or not this {@code FormHandlerParamField} is equal to another. Such equality
+     * depends only on the field's name (pattern), not it's value, enabled state, or regex state.
+     *
+     * <p>When the {@code FormHandlerParamField} is regex then a case sensitive match is done,
+     * otherwise a case insensitive match is done.
+     *
+     * @param obj the object to compare against.
+     * @return true if the objects are considered equal, false otherwise.
+     */
     @Override
     public boolean equals(Object obj) {
-        if (this == obj) return true;
```

Comment:
I think this should be retained.

Reply:
I'll test again but I think this was causing me issues with the `contains()` equality check.